src/containers/my-cooperations/cooperation-activities-list/CooperationActivitiesList.tsx

```diff
@@ -56,18 +56,42 @@ const CooperationActivitiesList = () => {
     setSectionsData(newSections)
   }
 
-  if (!data.sections.length && !selectedCourse) {
-    addNewSection()
-  }
+  useEffect(() => {
+    if (!data.sections.length && !isAddedClicked) {
+      addNewSection()
+    }
 
-  if (selectedCourse && !data.sections.length) {
-    const allSections = selectedCourse.sections.map((section, index) => ({
-      ...section,
-      id: Date.now().toString() + index
-    }))
+    if (selectedCourse && !data.sections.length && isAddedClicked) {
+      const allSections = selectedCourse.sections.map((section, index) => ({
+        ...section,
+        id: Date.now().toString() + index
+      }))
+      setSectionsData(allSections)
+    }
 
-    setSectionsData(allSections)
-  }
+    if (selectedCourse && data.sections.length && isAddedClicked) {
+      const addNewSectionsCourse = (index: number | undefined = undefined) => {
+        const newSectionData = selectedCourse.sections.map(
+          (section, index) => ({
+            ...section,
+            id: Date.now().toString() + index
+          })
+        )
+        let newSections = []
+        if (index !== undefined) {
+          newSections = [
+            ...data.sections.slice(0, index),
+            ...newSectionData,
+            ...data.sections.slice(index)
+          ]
+        } else {
+          newSections = [...data.sections, ...newSectionData]
+        }
+        setSectionsData(newSections)
+      }
+      addNewSectionsCourse(currentSectionIndex)
+    }
+  }, [isAddedClicked])
```

Comment:
Do you have dependencies issue?

Reply:
![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/101183408/0a3b7465-ffb4-404a-ab2f-f87369797659)

Yes, there is. 
but I can't change this implementation. 
I talked to Anatoly about it, he said it's not critical.